src/analyze/analyze-security.c

```diff
@@ -1560,8 +1564,27 @@ static int assess(const SecurityInfo *info, Table *overview_table, AnalyzeSecuri
                         (void) table_set_display(details_table, (size_t) 0, (size_t) 1, (size_t) 2, (size_t) 6);
         }
 
-        for (i = 0; i < ELEMENTSOF(security_assessor_table); i++) {
-                const struct security_assessor *a = security_assessor_table + i;
+        if (v != NULL) {
+                security_table_size = json_variant_elements(v)/2;
+                JsonVariant *val;
+
+                for (size_t key = 0; key < security_table_size; key++) {
+                        val = json_variant_by_key(v, security_table[key].id);
+                        assert_se(val && json_variant_type(val) == JSON_VARIANT_OBJECT && json_variant_elements(val) > 0);
+
+                        security_table[key].range = json_variant_unsigned(json_variant_by_key(val, "range"));
+                        security_table[key].weight = json_variant_unsigned(json_variant_by_key(val, "weight"));
+                        if (json_variant_by_key(val, "description_na"))
+                                security_table[key].description_na = json_variant_string(json_variant_by_key(val, "description_na"));
+                        if (json_variant_by_key(val, "description_bad"))
+                                security_table[key].description_bad = json_variant_string(json_variant_by_key(val, "description_bad"));
+                        if (json_variant_by_key(val, "description_good"))
+                                security_table[key].description_good = json_variant_string(json_variant_by_key(val, "description_good"));
+                }
+        }
```

Comment:
please don't mangle the table beforehand. Instead, provide small accessor functions for the relevant fileds of the table entry, that either returns the data in the static/constant table or in the json object if defined there. i.e. merge on request, not beforehand.